src/meta/client/MetaClient.cpp

```diff
@@ -1282,14 +1335,96 @@ MetaClient::getEdgeSchemaFromCache(GraphSpaceID spaceId, EdgeType edgeType, Sche
     }
 }
 
-StatusOr<const cpp2::IndexProperties>
-MetaClient::getTagIndexFromCache(TagIndexID) {
-    LOG(FATAL) << "Not implemented";
+StatusOr<std::shared_ptr<const cpp2::IndexFields>>
+MetaClient::getTagIndexFromCache(GraphSpaceID spaceId, TagIndexID tagIndexID) {
+    if (!ready_) {
+        return Status::Error("Not ready!");
+    }
+
+    folly::RWSpinLock::ReadHolder holder(localCacheLock_);
+    auto spaceIt = localCache_.find(spaceId);
+    if (spaceIt == localCache_.end()) {
+        // Not found
+        VLOG(3) << "Space " << spaceId << " not found!";
+        return Status::SpaceNotFound();
+    } else {
+        auto iter = spaceIt->second->tagIndexes_.find(tagIndexID);
+        if (iter == spaceIt->second->tagIndexes_.end()) {
+            VLOG(3) << "Space " << spaceId << ", Tag " << tagIndexID << " not found!";
+            return Status::TagNotFound();
+        } else {
+            return iter->second;
+        }
+    }
+}
+
+StatusOr<std::shared_ptr<const cpp2::IndexFields>>
+MetaClient::getEdgeIndexFromCache(GraphSpaceID spaceId, EdgeIndexID edgeIndexID) {
+    if (!ready_) {
+        return Status::Error("Not ready!");
+    }
+
+    folly::RWSpinLock::ReadHolder holder(localCacheLock_);
+    auto spaceIt = localCache_.find(spaceId);
+    if (spaceIt == localCache_.end()) {
+        // Not found
+        VLOG(3) << "Space " << spaceId << " not found!";
+        return Status::SpaceNotFound();
+    } else {
+        auto iter = spaceIt->second->edgeIndexes_.find(edgeIndexID);
+        if (iter == spaceIt->second->edgeIndexes_.end()) {
+            VLOG(3) << "Space " << spaceId << ", EdgeType " << edgeIndexID << " not found!";
+            return std::shared_ptr<const cpp2::IndexFields>();
+        } else {
+            return iter->second;
+        }
+    }
+}
+
+StatusOr<std::vector<std::shared_ptr<const cpp2::IndexFields>>>
+MetaClient::getTagIndexesFromCache(GraphSpaceID spaceId) {
+    if (!ready_) {
+        return Status::Error("Not ready!");
+    }
+
+    folly::RWSpinLock::ReadHolder holder(localCacheLock_);
+    auto spaceIt = localCache_.find(spaceId);
+    if (spaceIt == localCache_.end()) {
+        VLOG(3) << "Space " << spaceId << " not found!";
+        return Status::SpaceNotFound();
+    } else {
+        auto tagIndexes = spaceIt->second->tagIndexes_;
+        auto iter = tagIndexes.begin();
+        std::vector<std::shared_ptr<const cpp2::IndexFields>> fields;
+        while (iter != tagIndexes.end()) {
+            fields.emplace_back(iter->second);
+            iter++;
+        }
+        return fields;
+    }
```

Comment:
Where is indexId ?I think it's useful.

Reply:
what is Index ID doing？

Reply:
> what is Index ID doing？

scan index via indexID , or write index data using indexId.